pytorch_lightning/callbacks/swa.py

```diff
@@ -166,8 +168,20 @@ def on_train_epoch_start(self, trainer: 'pl.Trainer', pl_module: 'pl.LightningMo
                 last_epoch=trainer.max_epochs if self._annealing_strategy == "cos" else -1
             )
 
-            rank_zero_warn(f"Swapping lr_scheduler {lr_scheduler} for {self._swa_scheduler}")
-            trainer.lr_schedulers[0]["scheduler"] = self._swa_scheduler
+            if trainer.lr_schedulers:
+                lr_scheduler = trainer.lr_schedulers[0]["scheduler"]
+                rank_zero_warn(f"Swapping lr_scheduler {lr_scheduler} for {self._swa_scheduler}")
+                trainer.lr_schedulers[0]["scheduler"] = self._swa_scheduler
+            else:
+                trainer.lr_schedulers.append({
+                    'scheduler': self._swa_scheduler,
+                    'name': None,  # no custom name
+                    'interval': 'epoch',  # after epoch is over
+                    'frequency': 1,  # every epoch/batch
+                    'reduce_on_plateau': False,  # most often not ReduceLROnPlateau scheduler
+                    'monitor': None,  # value to monitor for ReduceLROnPlateau
+                    'strict': True,  # enforce that the monitor exists for ReduceLROnPlateau
+                })
```

Comment:
shall be taken from `configure_schedulers`'s `default_config` rather than defining it here again? just in case more key-value pairs are added to this default_dict in the future then we might need to update here again too.

Reply:
Something similar just got merged: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/plugins/training_type/deepspeed.py#L243

I wonder if we can define some form of wrapper? or just a `default_config` property that returns a dict, and we append the scheduler? This property could live in the optimizers.py

Reply:
SWALR shouldn't follow the user own scheduler as they are different. Not sure what you mean by default.

Reply:
yes @SeanNaren 

@tchaton what I mean here is the default values are taken from configure_scheduler's default config. It's just repetitive code. Let's say we change the default interval from epoch to something else then one might need to change it everywhere wherever it's defined. We can just refer to it here by assigning it somewhere as @SeanNaren suggested.

Reply:
Hey @rohitgr7, mind having a look ?